src/converters/shmup/mod.rs

```diff
@@ -58,26 +60,29 @@ fn build_converter_data(extracted_data: &extractor::Data) -> Data {
     let objs = &extracted_data.git.objects;
     let files = &extracted_data.code.files_data;
 
-    let mut data = Data {
+    let data = Data {
         ..Default::default()
     };
 
+    let mut_data = Arc::new(Mutex::new(data));
+
     //Get the initial commit
     if let Some(commit) = &objs[commit_oid].commit {
         let trees_oid = vec![commit.tree.clone()];
-        add_scenes(&trees_oid, &mut data, objs, files);
+        add_scenes(&trees_oid, mut_data.clone(), objs, files);
     }
 
+    let data = mut_data.lock().unwrap().to_owned();
     data
```

Comment:
Is it usefull to initialize the variable `data` line 63 to then change it's value line 75 and immediatly return it ?

Reply:
The `data` variable we initialize on line 63 is being heavily changed through a Mutex by the function `add_scenes` on line 106. The variable on line 75 is just used to shadow the initial data variable we could remove that shadow variable completely and return only `mut_data.lock().unwrap().to_owned()`, I will do that

Reply:
I actually can't do that because the `mut_data.lock()` creates temporary variable and we need to own that variable to be able to return it to the caller. The compiler output for context 🙇‍♂️ 

```console
   |
75 |     mut_data.lock().unwrap().to_owned()
   |     ^^^^^^^^^^^^^^^---------
   |     |
   |     borrowed value does not live long enough
   |     a temporary with access to the borrow is created here ...
76 | }
   | -
   | |
   | `mut_data` dropped here while still borrowed
   | ... and the borrow might be used here, when that temporary is dropped and runs the `Drop` code for type `std::sync::MutexGuard`
   |
   = note: the temporary is part of an expression at the end of a block;
           consider forcing this temporary to be dropped sooner, before the block's local variables are dropped
help: for example, you could save the expression's value in a new local variable `x` and then make `x` be the expression at the end of the block
   |
75 |     let x = mut_data.lock().unwrap().to_owned(); x
   |     +++++++                                    +++
```